### PR TITLE
Support new argument --disable-probe for app_create, app_update, and deployment_create in spring-cloud

### PR DESCRIPTION
**Pull request by CaihuaRui**
…eployment_create in spring-cloud

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring-cloud

File: src/spring-cloud/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Release History
 ===============
+2.12.2
```
Comment (wangzelin007) [inline src/spring-cloud/HISTORY.md:3]: Better to use 2.12.0 or 2.11.3.

File: src/spring-cloud/setup.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '2.11.2'
+VERSION = '2.12.2'
```
Comment (wangzelin007) [inline src/spring-cloud/setup.py:19]: same as above.

**Review by wangzelin007 — APPROVED** [review]
LGTM

File: src/spring-cloud/HISTORY.md (hunk 1 above)
Comment (CaihuaRui) [reply src/spring-cloud/HISTORY.md]: Updated

File: src/spring-cloud/setup.py (hunk 2 above)
Comment (CaihuaRui) [reply src/spring-cloud/setup.py]: Updated

**Review by ShichaoQiu — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**